source/traction_boundary_conditions/ascii_data.cc

```diff
@@ -0,0 +1,143 @@
+    {
+      prm.enter_subsection("Boundary traction model");
+      {
+        Utilities::AsciiDataBoundary<dim>::parse_parameters(prm);
+      }
+      prm.leave_subsection();
+    }
+
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace TractionBoundaryConditions
+  {
+    ASPECT_REGISTER_TRACTION_BOUNDARY_CONDITIONS(AsciiData,
+                                                 "ascii data",
+                                                 "Implementation of a model in which the boundary "
+                                                 "traction is derived from files containing pressure data "
+                                                 "in ascii format. The pressure given in the data file is "
+                                                 "applied as traction normal to the surface of a given boundary, "
+                                                 "pointing inwards. Note the required format of the "
+                                                 "input data: The first lines may contain any number of comments "
+                                                 "if they begin with '#', but one of these lines needs to "
+                                                 "contain the number of grid points in each dimension as "
+                                                 "for example '# POINTS: 3 3'. "
+                                                 "The order of the data columns "
+                                                 "has to be 'x', 'Pressure [Pa]' in a 2d model and "
+                                                 " 'x', 'y', 'Pressure [Pa]' in a 3d model, which means that "
+                                                 "there has to be a single column "
+                                                 "containing the pressure. "
+                                                 "Note that the data in the input "
+                                                 "files need to be sorted in a specific order: "
+                                                 "the first coordinate needs to ascend first, "
+                                                 "followed by the second in order to "
+                                                 "assign the correct data to the prescribed coordinates. "
+                                                 "If you use a spherical model, "
+                                                 "then the data will still be handled as Cartesian, "
+                                                 "however the assumed grid changes. 'x' will be replaced by "
+                                                 "the radial distance of the point to the bottom of the model, "
+                                                 "'y' by the azimuth angle and 'z' by the polar angle measured "
+                                                 "positive from the north pole. The grid will be assumed to be "
```

Comment:
Why do you think the radius can't be a column? If it is a spherical shell geometry model with an opening angle of <360°, we also need boundary conditions at the sides, see for example in `data/velocity-boundary-conditions/ascii-data/test/shell_3d_east.0.txt`